docs/metatype.dev/docs/reference/graphql/index.mdx

```diff
@@ -0,0 +1,183 @@
+
+The simple primitive typegraph types translate directly to their [GraphQL equivalent](https://graphql.org/learn/schema/#scalar-types).
+This includes standard scalar variants like `t.uuid` or any user declared aliases with custom configurations.
+No custom GraphQL scalars are generated for such types.
+
+
+| Type                       | GraphQL type | Description                                                                                                   |
+| -------------------------- | ------------ | ------------------------------------------------------------------------------------------------------------- |
+| `t.integer()` | `Int`        | Represents signed 32-bit integers.                                                                            |
+| `t.float()`     | `Float`      | Represents signed double-precision values as specified by [IEEE 754](https://en.wikipedia.org/wiki/IEEE_754). |
+| `t.boolean()` | `Boolean`    | Represents `true` or `false`.                                                                                 |
+| `t.string()`   | `String`     | Represents textual data as UTF-8 character sequences.                                                         |
+
+### Files
+
+File types, primarily used through the [`S3Runtime`](/docs/reference/runtimes/s3), translate to a custom scalar called `File` in the GraphQL schema.
+The `S3Runtime` provides a set of helpers for all your file needs including support for [multipart HTTP requests](/docs/guides/files-upload) for file upload.
+
+### Structs
+
+`t.structs` types translate to:
+- [GraphQL input types](https://graphql.org/learn/schema/#input-types) when used as inputs to functions
+- [GraphQL object types](https://graphql.org/learn/schema/#object-types-and-fields) in any other place.
+
+### Lists
+
+`t.list` types simply convert to [GraphQL list types](https://graphql.org/learn/schema/#lists-and-non-null).
+
+### Functions
+
+`t.func` types are represented by their output type in the GraphQL schema. 
+If their input struct has fields present, these are converted to [arguments](https://graphql.org/learn/schema/#arguments) for the field.
+
+### Unions and Eithers
+
+Unions and either translate to a number of different forms depending on their usage in the graph and their composition.
+
+When used as a field in an input struct, unions/eithers are converted to custom scalars since the GraphQL spec doesn't allow GraphQL unions on input types.
+These scalars expect the standard JSON representation of your value and that is all.
+
+When querying union/either values, there are a few more details.
+For struct variants, the standard GraphQL rules apply and [inline fragments](https://graphql.org/learn/queries/#inline-fragments) must be used.
+Currently, all struct variants must have inline fragments present even if the user is not interested in them.
+And as per GraphQL spec, common field querying is not supported and any fields must be part of the inline fragment. 
```

Comment:
Ticket?

Reply:
Did you mean to mark the line after where `__typename` is not supported?